File: solutions/javascript/isbn-verifier/1/isbn-verifier.js
export const isValid = (isbn) => {

  const cleanIsbn = isbn.replace(/-/g, '');
  
  if (!/^\d{9}[\dX]$/.test(cleanIsbn)) {
    return false;
  }


  const sum = [...cleanIsbn].reduce((acc, char, index) => {
    const value = char === 'X' ? 10 : parseInt(char, 10);
    return acc + value * (10 - index);
  }, 0);

  return sum % 11 === 0;
};

File: solutions/kotlin/isbn-verifier/1/src/main/kotlin/IsbnVerifier.kt
class IsbnVerifier {

    fun isValid(number: String): Boolean {

        val cleaned = number.replace("-", "")

 
        if (cleaned.length != 10) return false

        var sum = 0

        for (i in 0 until 10) {
            val char = cleaned[i]
            val digit = when {
                char.isDigit() -> char.digitToInt()
           
                char == 'X' && i == 9 -> 10
                else -> return false
            }

      
            val weight = 10 - i
            sum += digit * weight
        }

  
        return sum % 11 == 0
    }
}
